### PR TITLE
Add client to download files from open data server 

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
Signed-off-by: ChamseddineBhd <chamseddine.benhamed@ensi-uma.tn>

### CODE REVIEW

File: src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    @Override
+    public List<SubstationGeoData> getSubstations() {
+        ByteArrayInputStream byteArrayInputStream = downloadFile("/explore/dataset/postes-electriques-rte/download/");
+        LOGGER.info("substations were downloaded from the open data server");
+        return new ArrayList<>(GeographicDataParser.parseSubstations(new BufferedReader(new InputStreamReader(byteArrayInputStream))).values());
+    }
+
+    @Override
+    public List<LineGeoData> getLines() {
+        ByteArrayInputStream undergroundLinesByteArrayInputStream =  downloadFile("/explore/dataset/lignes-souterraines-rte/download/");
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java:81]: Is it better to make these URLs parametrable or we don't have any use case that requires it ?

File: src/main/java/org/gridsuite/odre/server/utils/GeographicDataParser.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+import org.supercsv.prefs.CsvPreference;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.*;
+
+import static java.util.Collections.min;
+import static java.util.Collections.reverse;
+
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/odre/server/utils/GeographicDataParser.java:25]: license header + author

File: src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ *     <li>
+ *         <a href="https://opendata.reseaux-energies.fr/explore/dataset/lignes-souterraines-rte/download/?format=csv&timezone=Europe/Berlin&use_labels_for_header=true">lignes-souterraines-rte.csv</a>
+ *     </li>
+ * </ul>
+ *
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+
+@Component
+public class OdreOpenDataClientImpl implements OdreClient {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java:44]: OD in ODRE already means Open Data.
OdreClientDownloadImpl ?

File: src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java (hunk 1 above)
Comment (ChamseddineBhd) [reply src/main/java/org/gridsuite/odre/server/client/OdreOpenDataClientImpl.java:81]: we don't have any use case that requires it

File: pom.xml [hunk 4]
```diff
@@ -80,6 +80,11 @@
 
     <dependencies>
         <!-- Compile dependencies -->
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.6</version>
```
Comment (geofjamg) [inline pom.xml:86]: It should be done with a property

**Pull request merged**